bin/rbenv-update

```diff
@@ -26,33 +27,57 @@ else
   reset=""
 fi
 
-indent_output() {
-  while read data; do
-    printf " ${color}|${reset}  %s\n" "$data"
-  done
+# Helper functions
+pretty_output() {
+  if [[ -n "$verbose" ]]; then
+    while read -r data; do
+      printf " ${color}|${reset}  %s\n" "$data"
+    done
+  else
+    cat >/dev/null 2>&1
+  fi
+}
+
+print_colored() {
+  if [[ -z "$silent" ]]; then
+    printf "${color}%s${reset}\n" "$1"
+  fi
+}
+
+git() {
+  if [[ -n "$noop" ]]; then
+    echo git "$@"
+  else
+    command git "$@" | pretty_output
+  fi
 }
 
 is_rbenv_git_repo() {
-  git remote -v 2>/dev/null | grep -q 'rbenv\|ruby-build'
+  command git remote -v 2>/dev/null | grep -q 'rbenv\|ruby-build'
 }
 
 rbenv_update() {
-  printf "${color}updating %s${reset}\n" "$1"
   if is_rbenv_git_repo; then
-    git pull --no-rebase --ff 2>&1 | indent_output
+    print_colored "Updating $1"
+    git pull --no-rebase --ff 2>&1
   else
-    echo "Not an rbenv git repo; skipping..." | indent_output
+    print_colored "Skipping $1 since it's not an rbenv git repo"
   fi
-  echo
 }
 
-cd "$(dirname $(which rbenv))"
+# Main script
+if [[ -n "$noop" && -n "$silent" ]]; then
```

Comment:
it's unnecessary to quote vars within `[[` (that's only necessary in `[`, because `[[` prevents word splitting)

Though this raises the general question of whether we should be using `[` alone. If I'm not mistaken, rbenv itself uses `[` almost exclusively to reach widest possible audience of shells. Perhaps we should do likewise and only use `[`? (I realize these weren't all introduced with your PR, but it did bring it to my attention.)

Wondering if someone else could weigh in? @rkh ?